File: internal/domain/slot.go
package domain

import "time"

type SlotStatus string

const (
	StatusAvailable SlotStatus = "available"
	StatusBooked    SlotStatus = "booked"
)

type Slot struct {
	ID        uint
	TimeSlot  string
	Status    SlotStatus
	BookedBy  string
	UpdatedAt time.Time
}

File: client/src/types/slot.ts
export interface Slot {
  ID: number
  TimeSlot: string
  Status: "available" | "booked"
  BookedBy: string
  UpdatedAt: string
}

export interface BookSlotRequest {
  slot_id: number
  user: string
}

export interface BookSlotResponse {
  message: string
  slot: Slot
}

export interface WsMessage {
  event: string
  data: {
    slot_id: number
  }
}
